### PR TITLE
syslog-ng-ctl: workaround invalid escaping in syslog-ng/AxoSyslog metrics

### PR DESCRIPTION
**Pull request by MrAnno**
Prior to the following patch, syslog-ng/AxoSyslog escaped control characters and `\` incorrectly.

https://github.com/axoflow/axosyslog/pull/224

Fixes axoflow/axoflow#2129

### CODE REVIEW

**Review by OverOrion — APPROVED** [review]

File: pkg/syslog-ng-ctl/stats_prometheus.go [hunk 1]
```diff
@@ -154,6 +154,39 @@ func transformEventDelayMetric(delayMetric *io_prometheus_client.MetricFamily, d
...
+			continue
+		}
+
+		if i+1 >= length {
+			fixedOutput.WriteString("\\\\")
+			break
+		}
+
+		next := output[i+1]
+		if next == '\\' || next == 'n' || next == '"' {
```
Comment (siliconbrain) [inline pkg/syslog-ng-ctl/stats_prometheus.go:177]: nit
```suggestion
		if next := output[i+1]; next == '\\' || next == 'n' || next == '"' {
```

File: pkg/syslog-ng-ctl/stats_prometheus.go [hunk 2]
```diff
@@ -154,6 +154,39 @@ func transformEventDelayMetric(delayMetric *io_prometheus_client.MetricFamily, d
...
+	for i := 0; i < length; i++ {
+		c := output[i]
+
+		if c != '\\' {
+			fixedOutput.WriteByte(c)
+			continue
+		}
+
+		if i+1 >= length {
+			fixedOutput.WriteString("\\\\")
```
Comment (siliconbrain) [inline pkg/syslog-ng-ctl/stats_prometheus.go:172]: could write these without escaping using backticks, but I'm not sure if that's less or more confusing 🤷
```suggestion
			fixedOutput.WriteString(`\\`)
```

**Review by siliconbrain — APPROVED** [review]
lgtm

File: pkg/syslog-ng-ctl/stats_prometheus.go (hunk 2 above)
Comment (MrAnno) [reply pkg/syslog-ng-ctl/stats_prometheus.go:172]: I think this looks slightly better :)

**Pull request merged**